src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp.js';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+  const timeEntry=props.timeStamp
+
+  const onLikeButtonClick = () => {
+    const updatedChat = {
+        id: props.id,
+        sender: props.sender,
+        body: props.body,
+        timeStamp: props.timeStamp,
+        liked: !props.liked
+    };
+
+    props.onUpdateChat(updatedChat);
+  };
+
+  const heartColor = props.liked ? '❤️' : '🤍';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={props.chatEntry}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={timeEntry}></TimeStamp></p>
+          <button className="like" onClick={onLikeButtonClick}>{heartColor}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool,
+  onUpdateChat: PropTypes.func.isRequired
```

Comment:
The PropType check is missing `id` and `chatEntry` that are passed into each ChatEntry component on lines 14 and 20 of ChatLog.js